tests/base.py

```diff
@@ -527,3 +554,68 @@ def run_initial_sync(self, environment, data_type):
                                                                          self.expected_primary_keys())
 
         return conn_id, first_record_count_by_stream
+
+    def assertRecordsEqualByPK(self, stream, expected_records, actual_records):
+        """Compare expected and actual records by their primary key."""
+
+        primary_keys = list(self.expected_primary_keys().get(stream)) if self.expected_primary_keys().get(stream) else self.makeshift_primary_keys().get(stream)
+
+        # Verify there are no duplicate pks in the target
+        actual_pks = [tuple(actual_record.get(pk) for pk in primary_keys) for actual_record in actual_records]
+        actual_pks_set = set(actual_pks)
+        self.assertEqual(len(actual_pks), len(actual_pks_set), msg="A duplicate record may have been replicated.")
+        actual_pks_to_record_dict = {tuple(actual_record.get(pk) for pk in primary_keys): actual_record for actual_record in actual_records}
+
+        # Verify there are no duplicate pks in our expectations
+        expected_pks = [tuple(expected_record.get(pk) for pk in primary_keys) for expected_record in expected_records]
+        expected_pks_set = set(expected_pks)
+        self.assertEqual(len(expected_pks), len(expected_pks_set), msg="Our expectations contain a duplicate record.")
+        expected_pks_to_record_dict = {tuple(expected_record.get(pk) for pk in primary_keys): expected_record for expected_record in expected_records}
+
+        # Verify that all expected records and ONLY the expected records were replicated
+        self.assertEqual(expected_pks_set, actual_pks_set)
+
+        return expected_pks_to_record_dict, actual_pks_to_record_dict
+
+    def assertRecordsEqual(self, stream, expected_record, sync_record):
+        if stream == 'payments':
+            self.assertDictEqualWithOffKeys(expected_record, sync_record, {'updated_at'})
+        elif stream in {'employees', 'roles'}:
+            self.assertDictEqualWithOffKeys(expected_record, sync_record, {'created_at', 'updated_at'})
+        elif stream == 'orders':
+            self.assertParentKeysEqual(expected_record, sync_record)
+            expected_record_copy = deepcopy(expected_record)
+            sync_record_copy = deepcopy(sync_record)
+
+            if 'fulfillments' in sync_record_copy:
+                expected_record_fulfillment = expected_record_copy['fulfillments'][0]
+                synced_record_fulfillment = sync_record_copy['fulfillments'][0]
+                self.assertParentKeysEqual(expected_record_fulfillment, synced_record_fulfillment)
+
+                if 'pickup_details' in synced_record_fulfillment:
+                    expected_record_pickup_details = expected_record_fulfillment['pickup_details']
+                    if 'expired_at' in expected_record_pickup_details:
+                        # No explanation for why this key exists in the expected_record_pickup_details but not in synced record's
```

Comment:
After thinking about this it's probably a missing field in the schema... great

Reply:
Confirmed the schema for pickup_details has no expired_at field... this one took a while to debug.